WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -446,6 +448,24 @@ def submitJobs( self ):
       platform = self.queueDict[queue]['Platform']
       siteMask = siteName in siteMaskList
 
+      # Check the status of the Site
+      result = self.sstClient.getSiteStatuses({siteName})
+      if result['Value']:
```

Comment:
Alright, but before doing that you should also check for `result['OK']`, otherwise you risk triggering an exception (`result['Value']` is not there if `not result['OK']`)